### ISSUE TITLE
unhashable and not useful WMS parameters are added to the WMS URL request

### ISSUE THREAD

**Issue by keul** [PaulLeCam/react-leaflet#626]
## Bug report in v2

- [x] All peer dependencies are installed: React, ReactDOM and Leaflet.
- [x] Using a supported version of React and ReactDOM (v16.3.0 minimum).
- [x] Using the supported version of Leaflet (v1.4.x) and its corresponding CSS file is loaded.
- [x] Using the [latest version of React-Leaflet](https://github.com/PaulLeCam/react-leaflet/releases) - **not v1.x**.
- [x] The issue has not already been reported.
- [x] Make sure you have followed the [quick start guide](http://leafletjs.com/examples/quick-start.html) for Leaflet.
- [x] Make sure you have fully read the [documentation](https://react-leaflet.js.org/docs/en/intro.html) and that you understand the [limitations](https://react-leaflet.js.org/docs/en/intro.html#limitations).

A similar issue (#373) has been already discussed and fixed.

Actually I see too many parameters added to the WMS request URL.

### Expected behavior

Do not automatically append every prop to the URL, or provide a way to white-list props to be used for parameter.

At least: only string/number/null parameters should be used

**Note**: this would be not standard I fear... the automatic URL usage seems a default behavor also for WMS leaflet.

### Actual behavior

A WMS URL can became quickly something like this:

```https://apps.ecmwf.int/wms/?&SERVICE=WMS&REQUEST=GetMap&LAYERS=background%2Cnamed_cyclones%2Cnamed_cyclones_tracks%2Cforeground&STYLES=&FORMAT=image%2Fpng&TRANSPARENT=true&VERSION=1.3&TOKEN=public&TILES=true&FOO=123%2C5566&CHILDREN=undefined&LEAFLET=%5Bobject%20Object%5D&WIDTH=256&HEIGHT=256&CRS=EPSG%3A4326&BBOX=45,67.5,67.5,90```

* If I pass a `children` to my WMSTileLayer it is added to the URL. Even if children is not used by WMSTileLayer it should be discarded
* a `LEAFLET=[object Object]` is always appended, this came from react-leaflet internal guess
* if I add a props like an array as `foo: [123, 5566]` should this be encoded and used?

### Steps to reproduce

https://codesandbox.io/s/react-leaflet-wms-params-issue-6y1k5

**Comment by PaulLeCam** [comment]
Adding the `leaflet` key is a mistake indeed, it will be fixed in the next release.
Regarding the `children`, this component doesn't support having children provided so that's up to your app.
The encoding of the parameters is handled by Leaflet so it's not up to React Leaflet.

**Comment by keul** [comment]
> The encoding of the parameters is handled by Leaflet so it's not up to React Leaflet.

Absolutely, I asked for this in the core leaflet.

> Regarding the children, this component doesn't support having children provided so that's up to your app.

I get you point but I not fully agree.

I've a code like this (simplified): 

```
let LayerType;
let children = undefined;

if (x) {
  LayerType = L.GeoJSON
  children = (
    <Tooltip>
      <span>something</span>
    </Tooltip>
  )
} else {
  LayerType = L.WMSTileLayer;
}

return (
  <LayerC
        {...props}
      >
        {children}
  </LayerC>
)
```

It seems that there's not way to pass an empty children (I tried `null` and `undefined`).

Just an opinion, but I think that RL should handle this. Of course: I can pass `children` inside props.

However: I can live with this. Thanks!

**Comment by PaulLeCam** [comment]
I get your point regarding `children`, though I'd rather not exclude it in case this is a valid parameter for the tiles URL.
I just released v2.5.0 with the removed `leaflet` parameter, please let me know if you still have the issue after upgrading.
